src/server/pfs/server/driver.go

```diff
@@ -2550,6 +2550,75 @@ func (d *driver) getFile(pachClient *client.APIClient, file *pfs.File, offset in
 	return grpcutil.NewStreamingBytesReader(getBlocksClient, nil), nil
 }
 
+func (d *driver) getFiles(pachClient *client.APIClient, objReader io.Reader, file *pfs.File, f func(*pfs.GetFileResponse) error) (retErr error) {
+	var buf bytes.Buffer
+	parseFilesFromStream := func(p string, node *hashtree.NodeProto) error {
+		for i := int64(0); i <= node.SubtreeSize; i += int64(grpcutil.MaxMsgSize) {
+			_, err := io.CopyN(&buf, objReader, int64(node.SubtreeSize))
+			if err != nil && err != io.EOF {
+				return err
+			}
+			gfr := &pfs.GetFileResponse{
+				Value: buf.Bytes(),
+			}
+			buf.Reset()
+			if i == 0 {
+				gfr.File = file
+				gfr.File.Path = p
+			}
+			err = f(gfr)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	commitInfo, err := d.inspectCommit(pachClient, file.Commit, pfs.CommitState_STARTED)
+	if err != nil {
+		fmt.Println("commit inspect err", err)
+		return err
+	}
+	// Handle commits to input repos
```

Comment:
Not a huge fan of all this copy/pasting from `getFile()` it's necessary here, because we need separate logic to handle input and output repos. Let me know if there is a better way to do this.